src/types/Property/property.ts

```diff
@@ -15,26 +15,26 @@ type EmptyObject = Record<string, never>;
  * Extracts the `additionalData` property from a specific property type in `PropertyTypes`.
  *
  * This utility type checks if the specified property type in `PropertyTypes` contains
- * an `additionalData` field. If it does, it creates a new type with the `additionalData`
- * field. Otherwise, it defaults to an empty object type.
+ * an `additionalData` field. If it does, it uses the inferred type for the additional
+ * data. Otherwise, it defaults to an empty object type.
  */
 type AdditionalData<T extends keyof PropertyTypes> = PropertyTypes[T] extends {
   additionalData: infer A;
 }
-  ? { additionalData: A }
+  ? A
   : EmptyObject;
```

Comment:
I was testing some stuff and a way to make typescript give an error when using a key that doesn't exist is to use {} instead of Record<string, never> for the EmptyObject